modules/core/src/main/scala/ciris/UseOnceSecret.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017-2021 Viktor Lövgren
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+package ciris
+
+import cats.effect.kernel.{Resource, Sync}
+import cats.implicits._
+import java.util.Arrays
+import java.util.concurrent.atomic.AtomicReference
+
+trait UseOnceSecret {
+  def resource[F[_]](implicit F: Sync[F]): Resource[F, Array[Char]]
+
+  final def useOnce[F[_], A](f: Array[Char] => F[A])(implicit F: Sync[F]): F[A] =
+    resource[F].use(f)
+}
+
+object UseOnceSecret {
+  final def apply[F[_]](secret: Array[Char])(implicit F: Sync[F]): F[UseOnceSecret] =
```

Comment:
would be nice to have a test showing error when reusing an used UseOnceSecret

Reply:
See 24d4315.